frontend/app-development/layout/SettingsModalButton/SettingsModal/components/Tabs/AccessControlTab/AccessControlTab.tsx

```diff
@@ -70,30 +85,54 @@ export const AccessControlTab = ({ org, app }: AccessControlTabProps): ReactNode
         );
       }
       case 'success': {
-        const currentPartyTypesAllowed = appMetadata?.partyTypesAllowed ?? initialPartyTypes;
+        const isNoCheckboxesChecked = Object.values(appMetadata.partyTypesAllowed).every(
+          (value) => !value,
+        );
+        const areAllCheckboxesChecked = Object.values(appMetadata.partyTypesAllowed).every(
+          (value) => value,
+        );
+        const isSomeCheckboxesChecked = Object.values(appMetadata.partyTypesAllowed).some(
+          (value) => value,
```

Comment:
These can also be moved to the SelectAllowedPartyTypes component

Reply:
fixed